cmd/root.go

```diff
@@ -0,0 +1,134 @@
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra/doc"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/k0sproject/k0s/pkg/build"
+	"github.com/k0sproject/k0s/pkg/util"
+)
+
+var cfgFile string
+var debug bool
+var cmdLogLevels map[string]string
+var logging map[string]string
+
+var defaultLogLevels = map[string]string{
+	"etcd":                    "info",
+	"containerd":              "info",
+	"konnectivity-server":     "1",
+	"kube-apiserver":          "1",
+	"kube-controller-manager": "1",
+	"kube-scheduler":          "1",
+	"kubelet":                 "1",
+}
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&cfgFile, "config", "c", "", "config file (default is $HOME/k0s.yaml)")
+	rootCmd.PersistentFlags().BoolVarP(&debug, "debug", "d", false, "Debug logging (default is false)")
+	rootCmd.PersistentFlags().StringToStringVarP(&cmdLogLevels, "logging", "l", defaultLogLevels, "Logging Levels for the different components")
```

Comment:
Again, why separately set here and not part of the command object creation?

Reply:
these are global flags, so they don't have to be set separately for each command.

Reply:
yep. should've read better the cobra way of defining these before commenting :)